curvefs/src/metaserver/s3compact_manager.cpp

```diff
@@ -208,16 +211,16 @@ void S3CompactManager::Enqueue() {
         auto copysetNode = CopysetNodeManager::GetInstance().GetCopysetNode(
             pinfo.poolid(), pinfo.copysetid());
         // traverse inode container
-        auto inodeStorage = s3compact->GetMutableInodeStorage();
-        InodeStorage::ContainerType inodes(inodeStorage->GetContainerData());
+        auto inodeManager = s3compact->GetInodeManager();
+        InodeStorage::ContainerType inodes(inodeManager->GetInodemap());
```

Comment:
The  function "inodeManager->GetInodemap" will copy all of the inode, if inodes are large, it will cost plenty of time, and there is a lock in inodestorge which will make inodestorage stall.

Reply:
current we need a copy of current inodemap to traverse, a pointer is not safe.
maybe inodeManager/inodeStorage can provide a thread safe traverse/next() method to me.